examples/intRVFL.py

```diff
@@ -0,0 +1,539 @@
+        "Seeds",
+        "Semeion",
+        "Soybean",
+        "Spambase",
+        "Spect",
+        "Spectf",
+        "StatlogAustralianCredit",
+        "StatlogGermanCredit",
+        "StatlogHeart",
+        "StatlogImage",
+        "StatlogLandsat",
+        "StatlogShuttle",
+        "StatlogVehicle",
+        "SteelPlates",
+        "SyntheticControl",
+        "Teaching",
+        "Thyroid",
+        "TicTacToe",
+        "Titanic",
+        "Trains",
+        "Twonorm",
+        "VertebralColumn2Clases",
+        "VertebralColumn3Clases",
+        "WallFollowing",
+        "Waveform",
+        "WaveformNoise",
+        "Wine",
+        "WineQualityRed",
+        "WineQualityWhite",
+        "Yeast",
+        "Zoo",
+    ]
+
+    def __init__(
+        self,
+        model_class: nn.Module,
+        simul: int = 1,
+        batch_size: int = 1,  # DK: The current code will not work for other values. Because "encoding_density_clipped()" assumes only one sample. Do we want to make it more generic?
+        device: torch.device = torch.device("cpu"),
+    ):
+        self.model_class = model_class
+        self.simul = simul
+        self.batch_size = batch_size
+        self.device = device
+
+        # Collects average accuracy for each simulation of each dataset
+        self.accuracy_collection = torch.zeros(
+            len(self.dataset_collection),
+            self.simul,
+            device=self.device,
+        )
+
+    ## Function compute the accuracy for a given model and data
+    def compute_accuracy(self, model, train_ds, test_ds):
+        # Make data loaders
+        train_ld = torch.utils.data.DataLoader(
+            train_ds, batch_size=self.batch_size, shuffle=True
+        )
+        test_ld = torch.utils.data.DataLoader(
+            test_ds, batch_size=self.batch_size, shuffle=True
+        )
+
+        # DK: I think this needs to be somehow connected to transform (callable, optional) that we have for datasets
+        # Used to make min-max normalization of the data
+        min_val = torch.min(train_ds.data, 0).values.to(self.device)
+        max_val = torch.max(train_ds.data, 0).values.to(self.device)
+
+        # Obtain the classifier for the model
+        model.fit(train_ld, min_val, max_val)
+
+        accuracy = torchmetrics.Accuracy()
+        with torch.no_grad():
+            for samples, labels in tqdm(test_ld, desc="Testing"):
+                samples = samples.to(self.device)
+
+                # DK: I think this needs to be somehow connected to transform (callable, optional) that we have for datasets
+                # Normalize
+                samples = (samples - min_val) / (max_val - min_val)
+
+                # Get values for each class
+                outputs = model(samples)
+                # Make prediction
+                # DK: Point for extension. Currently, it assumes implicitly that there is only one vector (prototype) per class, which does not have to be the case in general
```

Comment:
I think it makes sense if that would be handled in the model's forward path. That is, after you get a prediction for all your prototypes the classifier has to decide which class is selected by setting the output of that class as the max value. This way your current implementation is perfectly fine.

Reply:
This would make sense. Following this logic, should not we move line "predictions = torch.argmax(outputs, dim=-1)" inside model's forward path. I have done it in the revised code but happy to revert.

Reply:
We should try not to worry about these kinds of model details, see also my comment on a minimal version 1 design.